src/screens/WalletConnect/CallRequest/TransactionRequestContent.js

```diff
@@ -170,11 +170,31 @@ const useTransactionFee = (request: WalletConnectCallRequest) => {
   };
   const hasNotEnoughGas = !isEnoughBalanceForTransactionFee(walletBalances, balanceCheckTransaction, chain);
 
-  const { estimateCallRequestTransaction } = useWalletConnect();
+  const { estimateCallRequestTransaction, callRequests } = useWalletConnect();
 
-  React.useEffect(() => estimateCallRequestTransaction(request), [request, estimateCallRequestTransaction]);
+  React.useEffect(() => {
+    // perform additional check to avoid estimate on request dismiss
+    const requestExists = callRequests.some(({ callId }) => +callId === +request.callId);
```

Comment:
Why are we doing `+string` casting to `number` here? Could we go with regular string comparison?

```suggestion
    const requestExists = callRequests.some(({ callId }) => +callId === +request.callId);
```

Reply:
Not sure, definitely doesn't make sense. And it's actually a number heh 👀  changed